src/Sylius/Bundle/CoreBundle/EventListener/AvatarUploaderListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Core\Uploader\ImageUploaderInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Webmozart\Assert\Assert;
+
+final class AvatarUploaderListener
+{
+    /** @var ImageUploaderInterface */
+    private $uploader;
+
+    public function __construct(ImageUploaderInterface $uploader)
+    {
+        $this->uploader = $uploader;
+    }
+
+    public function uploadAvatar(GenericEvent $event): void
+    {
+        $subject = $event->getSubject();
+        Assert::isInstanceOf($subject, AdminUserInterface::class);
```

Comment:
Maybe you should add `ImageAwareInterface` and add it to `AdminUserInterface `, thanks to that, this listener could be more generic (similar as `ImagesUploadListener` for collection of images)

wdyt @Zales0123 @pamil @lchrusciel ?

Reply:
I have been added Avare interface but @pamil suggested that it is redundant